### PR TITLE
list available network interfaces

### PR DESCRIPTION
**Pull request by glmnet**
Just list the networks interfaces and their ip addresses, very useful in windows as finding the GUIDs are hard

### CODE REVIEW

File: ap2-receiver.py [hunk 1]
```diff
@@ -846,6 +846,16 @@ def upgrade_to_encrypted(self, client_address, shared_key):
                 exit(-1)
     except Exception:
```
Comment (LewdNeko) [inline ap2-receiver.py:847]: Why wait on an Exception? Make it a utility function, then add an optional argument to display it as standalone. Should make things easier

Comment (systemcrash) [reply ap2-receiver.py:847]: @glmnet can you separate this into a different commit (and optionally, PR)? Then we can focus on just the actual changes regarding the subject

File: ap2-receiver.py [hunk 2]
```diff
@@ -480,7 +480,7 @@ def do_SET_PARAMETER(self):
         elif content_type == HTTP_CT_IMAGE:
             if content_len > 0:
                 fname = None
-                with tempfile.NamedTemporaryFile(prefix="artwork", dir=".", delete=False) as f:
+                with tempfile.NamedTemporaryFile(prefix="artwork", dir=".", delete=False, suffix=".jpg") as f:
```
Comment (systemcrash) [inline ap2-receiver.py:483]: Move to separate commit plz

File: ap2-receiver.py [hunk 3]
```diff
@@ -846,6 +846,16 @@ def upgrade_to_encrypted(self, client_address, shared_key):
                 exit(-1)
     except Exception:
         print("[!] Network interface not found")
+        print("Available network interfaces:")
```
Comment (systemcrash) [inline ap2-receiver.py:849]: Agree - look at e.g. args.netiface, and if it's absent, call this code block there (also).
Call can also be in the exception block, altho from a readability perspective, it might not be immediately clear under which conditions the exception is triggered and reached.

**Comment by glmnet** [comment]
branch updated, had to make parameters optional

File: ap2-receiver.py [hunk 4]
```diff
@@ -480,7 +480,7 @@ def do_SET_PARAMETER(self):
         elif content_type == HTTP_CT_IMAGE:
             if content_len > 0:
                 fname = None
-                with tempfile.NamedTemporaryFile(prefix="artwork", dir=".", delete=False) as f:
+                with tempfile.NamedTemporaryFile(prefix="artwork", suffix=".jpg", dir=".", delete=False) as f:
```
Comment (systemcrash) [inline ap2-receiver.py:483]: Can we move this to a different PR please? (it's not related to listing network interfaces ;)
Airplay does not convert media types (to e.g. send only a jpg), and the receiver must be able to decode multiple file types. Most pics are probably sensible as a jpg, but there is media out there which is not jpg...

**Review by systemcrash — COMMENTED** [review]
👍

File: ap2-receiver.py (hunk 4 above)
Comment (glmnet) [reply ap2-receiver.py:483]: Content type is jpg not image

I miss the different pr comment. 

Do you squash merge?

**Comment by systemcrash** [comment]
I can retain all commits, I think.

But this idea can be expanded. I don't really see a point in writing images
to disk anyway. I think this can be made an optional command flag to retain
them. Otherwise don't even bother writing to disk.

**Comment by systemcrash** [comment]
Nice :)

File: ap2-receiver.py (hunk 4 above)
Comment (systemcrash) [reply ap2-receiver.py]: Sorry, I forgot that the `HTTP_CT_IMAGE` var is set to jpg. 🤦  But airplay is designed to receive other types, if I recall correctly.
